x/protorev/client/cli/utils.go

```diff
@@ -0,0 +1,245 @@
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	flag "github.com/spf13/pflag"
+
+	"github.com/osmosis-labs/osmosis/v15/x/protorev/types"
+)
+
+// ------------ types/functions to handle a SetHotRoutes CLI TX ------------ //
+type Trade struct {
+	Pool     uint64 `json:"pool"`
+	TokenIn  string `json:"token_in"`
+	TokenOut string `json:"token_out"`
+}
+
+type ArbRoutes struct {
+	Trades   []Trade `json:"trades"`
+	StepSize uint64  `json:"step_size"`
+}
+
+type hotRoutesInput struct {
+	TokenIn   string      `json:"token_in"`
+	TokenOut  string      `json:"token_out"`
+	ArbRoutes []ArbRoutes `json:"arb_routes"`
+}
+
+type createArbRoutesInput []hotRoutesInput
+
+type XCreateHotRoutesInputs hotRoutesInput
+
+type XCreateHotRoutesExceptions struct {
+	XCreateHotRoutesInputs
+	Other *string // Other won't raise an error
+}
+
+// UnmarshalJSON should error if there are fields unexpected.
+func (release *createArbRoutesInput) UnmarshalJSON(data []byte) error {
+	var createHotRoutesE []XCreateHotRoutesExceptions
+	dec := json.NewDecoder(bytes.NewReader(data))
+	dec.DisallowUnknownFields() // Force
+
+	if err := dec.Decode(&createHotRoutesE); err != nil {
+		return err
+	}
+
+	routes := make([]hotRoutesInput, 0)
+	for _, route := range createHotRoutesE {
+		routes = append(routes, hotRoutesInput(route.XCreateHotRoutesInputs))
+	}
+
+	*release = createArbRoutesInput(routes)
+	return nil
+}
+
+// extractTokenPairArbRoutes builds all of the TokenPairArbRoutes that were extracted from the json file
+func (release *createArbRoutesInput) extractTokenPairArbRoutes() []types.TokenPairArbRoutes {
+	if release == nil {
+		return nil
+	}
+
+	tokenPairArbRoutes := make([]types.TokenPairArbRoutes, 0)
+
+	// Iterate through each hot route and construct the token pair arb routes
+	for _, hotRoute := range *release {
+		current := types.TokenPairArbRoutes{}
+		current.TokenIn = hotRoute.TokenIn
+		current.TokenOut = hotRoute.TokenOut
+
+		for _, arbRoute := range hotRoute.ArbRoutes {
+			currentArbRoute := types.Route{}
+			currentArbRoute.StepSize = sdk.NewIntFromUint64(arbRoute.StepSize)
+
+			for _, trade := range arbRoute.Trades {
+				currentTrade := types.Trade{}
+				currentTrade.Pool = trade.Pool
+				currentTrade.TokenIn = trade.TokenIn
+				currentTrade.TokenOut = trade.TokenOut
+				currentArbRoute.Trades = append(currentArbRoute.Trades, currentTrade)
+			}
+
+			current.ArbRoutes = append(current.ArbRoutes, currentArbRoute)
+		}
+
+		tokenPairArbRoutes = append(tokenPairArbRoutes, current)
+	}
+
+	return tokenPairArbRoutes
+}
```

Comment:
Not sure if there is a more optimal way of reading in the json file and building the `TokenPairArbRoutes` since the only thing that is problematic is dealing with the StepSize which is an `sdk.Int`.

Reply:
The only other example I know of parsing a file withing our cli is the pool file:
https://github.com/osmosis-labs/osmosis/blob/3cd13a674a7647a4b3f48fad0c1fa7091a66b167/x/gamm/client/cli/parse.go#L27-L39

This implementation seems to be aligned with that. If parsing files becomes a common theme, we can better abstract it in the future